src/mapgen_functions.cpp

```diff
@@ -2629,6 +2630,32 @@ void mapgen_generic_house(map *m, oter_id terrain_type, mapgendata dat, int turn
             }
         }
         m->place_items("rare", 60, 0, 0, SEEX * 2 - 1, SEEY * 2 - 1, false, turn);
+
+    } else if (one_in(150)) {
+        for (int i = 0; i < SEEX * 2; i++) {
+            for (int j = 0; j < SEEY * 2; j++) {
+                if (m->ter(i, j) == t_floor) {
+                    if (one_in(15)) {
+                        m->add_spawn(mon_spider_cellar_giant, rng(1, 2), i, j);
+                        for (int x = i - 1; x <= i + 1; x++) {
+                            for (int y = j - 1; y <= j + 1; y++) {
+                                if (m->ter(x, y) == t_floor) {
+                                    madd_field( m, x, y, fd_web, rng(2, 3));
+                                    if (one_in(4)){
+                                     m->furn_set(i, j, f_egg_sackcs);
+                                     m->remove_field({i, j, m->get_abs_sub().z}, fd_web);
+                                    }
+                                }
+                            }
+                        }
+                    } else if (m->passable(i, j) && one_in(5)) {
+                        madd_field( m, x, y, fd_web, 1);
+                    }
+                }
+            }
+        }
+        m->place_items("rare", 60, 0, 0, SEEX * 2 - 1, SEEY * 2 - 1, false, turn);
```

Comment:
Is that entire block a total copy+paste of the block above?
If so, you should just roll spider and egg type at start of the block (just after `} else if( one_in( 150 ) ) {`) and then use that.
Like this:
```
auto spider_type = mon_spider_widow_giant;
auto egg_type = f_egg_sackbw;
if( one_in(2) ) {
    spider_type = mon_spider_cellar_giant;
    egg_type = f_egg_sackcs;
}
```

Then later in the function, instead of `m->add_spawn(mon_spider_cellar_giant, rng(1, 2), i, j);` you'd use `m->add_spawn(spider_type, rng(1, 2), i, j);` and instead of `m->furn_set(i, j, f_egg_sackcs);` use `m->furn_set(i, j, egg_type);`